### ISSUE TITLE
network tests fail on FreeBSD 13.0 in Cirrus-CI due to GCP image enabling TCP/UDP blackhole by default

### ISSUE THREAD

**Issue by emaste** [python/cpython#91824]
**Bug report**

Test fails with:
```
======================================================================
FAIL: test_imap4_host_default_value (test.test_imaplib.TestImaplib.test_imap4_host_default_value)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/Lib/test/test_imaplib.py", line 94, in test_imap4_host_default_value
    self.assertIn(cm.exception.errno, expected_errnos)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 60 not found in [61, 51, 49, 65]
```

(Cirrus-CI link: https://cirrus-ci.com/task/6657098355310592?logs=test#L26)

The list of accepted errnos comes from `expected_errnos = socket_helper.get_socket_conn_refused_errs()`

These are:
```
     49 EADDRNOTAVAIL Can't assign requested address. Normally results from an
             attempt to create a socket with an address not on this machine.

     51 ENETUNREACH Network is unreachable. A socket operation was attempted
             to an unreachable network.

     61 ECONNREFUSED Connection refused. No connection could be made because
             the target machine actively refused it.  This usually results
             from trying to connect to a service that is inactive on the
             foreign host.

     65 EHOSTUNREACH No route to host. A socket operation was attempted to an
             unreachable host.
```

but the test failed because the errno we observed is
```
     60 ETIMEDOUT Operation timed out. A connect(2) or send(2) request failed
             because the connected party did not properly respond after a
             period of time.  (The timeout period is dependent on the
             communication protocol.)
```

**Your environment**

- CPython versions tested on: git @ ac4ffd3be2049591701fee9f054191cce5ab9762
- Operating system and architecture: Cirrus-CI FreeBSD 13.0-RELEASE image

It may be that a Cirrus-CI firewall configuration results in the timeout (I will check), but depending on the purpose of a test perhaps `ETIMEDOUT` is a reasonable connection refused case

**Comment by emaste** [comment]
test_socket fails similarly,

```
======================================================================
FAIL: test_connect (test.test_socket.NetworkConnectionNoServer.test_connect)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/Lib/test/test_socket.py", line 5154, in test_connect
    self.assertEqual(cm.exception.errno, errno.ECONNREFUSED)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 60 != 61
======================================================================
FAIL: test_create_connection (test.test_socket.NetworkConnectionNoServer.test_create_connection)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/Lib/test/test_socket.py", line 5178, in test_create_connection
    self.assertIn(cm.exception.errno, expected_errnos)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 60 not found in [61, 51, 49, 65]
```

**Comment by emaste** [comment]
After disabling blackhole these tests pass

In .cirrus.yml
```
  network_setup_script:
    - sysctl net.inet.tcp.blackhole=0
    - sysctl net.inet.udp.blackhole=0
```

I should see about adding a check of these sysctls and having the tests skip if set.

**Comment by vstinner** [comment]
Issue fixed by commit a52a3509770f29f940cda9307704908949912276. Thanks for the bug report.
